pkg/build/builder/docker.go

```diff
@@ -305,7 +310,8 @@ func (d *DockerBuilder) dockerBuild(dir string, tag string, secrets []buildapi.S
 		NoCache:        noCache,
 		Pull:           forcePull,
 		BuildArgs:      buildArgs,
-		NetworkMode:    string(getDockerNetworkMode()),
+		NetworkMode:    network,
+		BuildBinds:     fmt.Sprintf("[\"%s:/etc/resolv.conf\"]", resolvConfHostPath),
```

Comment:
If I can inject \: into the resolve conf host path, can I break this?

Reply:
it'll break, i don't know that it'll get you anywhere.

if you can inject some quotes and a comma along with the colon you could presumably do pretty interesting things in terms of bindmounting host content into your build container.

I'm not sure how a user would have any control over that path though.

Regardless, we can certainly audit/strip those characters from the resolvConfHostPath.